tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/TraceAnnotations/TraceAnnotationsIntegration.cs

```diff
@@ -0,0 +1,109 @@
+// <copyright file="TraceAnnotationsIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Reflection;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations
+{
+    /// <summary>
+    /// Calltarget instrumentation to generate a span for any arbitrary method
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "",
+        TypeName = "",
+        MethodName = "",
+        ReturnTypeName = "",
+        ParameterTypeNames = new string[0],
+        MinimumVersion = "*",
+        MaximumVersion = "*",
+        IntegrationName = "TraceAnnotations")]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class TraceAnnotationsIntegration
+    {
+        private static readonly ConcurrentDictionary<RuntimeHandleTuple, Lazy<TraceAnnotationInfo>> InstrumentedMethodCache = new ConcurrentDictionary<RuntimeHandleTuple, Lazy<TraceAnnotationInfo>>();
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="methodHandle">The RuntimeMethodHandle representing the instrumented method</param>
+        /// <param name="typeHandle">The RuntimeTypeHandle representing the instrumented method's owning type</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle)
+        {
+            var info = InstrumentedMethodCache.GetOrAdd(
+                    new RuntimeHandleTuple(methodHandle, typeHandle),
+                    key => new Lazy<TraceAnnotationInfo>(() => TraceAnnotationInfoFactory.Create(MethodBase.GetMethodFromHandle(key.MethodHandle, key.TypeHandle))))
```

Comment:
I doubt the Lazy is useful here. `TraceAnnotationInfoFactory.Create` returns a struct and `MethodBase.GetMethodFromHandle` is reasonably cheap, so it's not an issue if multiple threads execute it on the first call.

Reply:
Makes sense, removed the Lazy in ac173c08b1b9c3c39876e5b55ae9edacc90bbf39